script/testing/micro_bench/run_micro_bench.py

```diff
@@ -0,0 +1,233 @@
+from micro_bench.config import Config
+from reporting.report_result import report_microbenchmark_result
+from micro_bench.constants import (JENKINS_URL, LOCAL_REPO_DIR, BENCHMARK_THREADS,
+                                   BENCHMARK_LOGFILE_PATH, BENCHMARK_PATH, MIN_REF_VALUES)
+from util.constants import LOG, PERFORMANCE_STORAGE_SERVICE_API
+
+
+def send_results(config, artifact_processor):
+    """ Iterate over the microbenchmark results to generate a comparison
+    against the histroical artifacts and send that comparrison to the 
+    performance storage service """
+    ret_code = 0
+    for bench_name in sorted(config.benchmarks):
+        filename = "{}.json".format(bench_name)
+        gbench_run_results = GBenchRunResult.from_benchmark_file(filename)
+
+        for key in sorted(gbench_run_results.benchmarks.keys()):
+            result = gbench_run_results.benchmarks.get(key)
+            LOG.debug("%s Result:\n%s", bench_name, result)
+
+            comparison = artifact_processor.get_comparison_for_publish_result(bench_name, result, config.lax_tolerance)
+            try:
+                report_microbenchmark_result(
+                    config.publish_results_env, result.timestamp, config, comparison)
+            except Exception as err:
+                LOG.error("Error reporting results to performance storage service")
+                LOG.error(err)
+                ret_code = 1
+
+    return ret_code
+
+
+def table_dump(config, artifact_processor):
+    """ Iterate over the microbenchmark results to generate a formatted table
+    to print out at the end of the test """
+    text_table = TextTable()
+    ret = 0
+    for bench_name in sorted(config.benchmarks):
+        filename = "{}.json".format(bench_name)
+        gbench_run_results = GBenchRunResult.from_benchmark_file(filename)
+
+        for key in sorted(gbench_run_results.benchmarks.keys()):
+            result = gbench_run_results.benchmarks.get(key)
+            LOG.debug("%s Result:\n%s", bench_name, result)
+
+            comparison = artifact_processor.get_comparison(
+                bench_name, result, config.lax_tolerance)
+            if comparison.get('pass') == 'FAIL':
+                ret = 1
+            text_table.add_row(comparison)
+
+    text_table.add_column("status")
+    text_table.add_column("iterations")
+    text_table.add_column("throughput", col_format="{:1.4e}")
+    text_table.add_column(
+        "ref_throughput", heading="ref throughput", col_format="{:1.4e}")
+    text_table.add_column("tolerance", heading="%tolerance")
+    text_table.add_column("change", heading="%change", col_format="{:+.0f}")
+    text_table.add_column("coef_var", heading="%coef var", col_format="{:.0f}")
+    text_table.add_column("reference_type", heading="ref type")
+    text_table.add_column("num_results", heading="#results")
+    text_table.add_column("suite")
+    text_table.add_column("test")
+    print("")
+    print(text_table)
+
+    return (ret)
```

Comment:
This is hardcore... Maybe you can try [PrettyTable](https://ptable.readthedocs.io/en/latest/tutorial.html) or [TextTables](https://texttables.readthedocs.io/en/latest/) or some other tools. This function and the `TextTable` class are magnificant.

Reply:
Great idea I'll look into how easy it would be to transition

Reply:
No need to change now I think, as long as it is working. I am just saying in the future maybe we can try those tools. I guess this `TextTable` class was legacy code inherited from the past right?